include/atomic.h

```diff
@@ -66,6 +71,8 @@ static inline atomic_val_t atomic_add(atomic_t *target, atomic_val_t value)
 {
 	return __atomic_fetch_add(target, value, __ATOMIC_SEQ_CST);
 }
+#elif defined(CONFIG_ATOMIC_OPERATIONS_C)
+__syscall atomic_val_t atomic_add(atomic_t *target, atomic_val_t value);
```

Comment:
You could do a local "OPCALL" or something thing here to avoid all the elif/else for all the calls just to declare exactly the same C function with either __sycall or extern.

Reply:
I like it. revision incoming.

Reply:
Wait, no that won't work.
This hopelessly confuses the code which scans for syscalls as it doesn't invoke the preprocessor.